portal-web/test/functional/com/liferay/portalweb/macros/SearchPortlets.macro

```diff
@@ -30,26 +30,43 @@ definition {
 	}
 
 	macro addWidgets {
-		if (IsNotVisible(locator1 = "ControlMenuAddPanel#CLOSE_ADD_PANEL")) {
-			Click.clickAt(locator1 = "ControlMenu#ADD");
-		}
+		if (${addWidgetViaUI} == "true") {
+			if (IsNotVisible(locator1 = "ControlMenuAddPanel#CLOSE_ADD_PANEL")) {
+				Click.clickAt(locator1 = "ControlMenu#ADD");
+			}
+
+			AssertElementPresent(locator1 = "ControlMenuAddPanel#SIDEBAR_HEADER");
 
-		AssertElementPresent(locator1 = "ControlMenuAddPanel#SIDEBAR_HEADER");
+			Navigator.gotoNavTab(navTab = "Widgets");
 
-		Navigator.gotoNavTab(navTab = "Widgets");
+			Type.sendKeysApplicationSearch(
+				locator1 = "NavBar#APPLICATION_SEARCH_FIELD",
+				value1 = "search");
 
-		Type.sendKeysApplicationSearch(
-			locator1 = "NavBar#APPLICATION_SEARCH_FIELD",
-			value1 = "search");
+			for (var portletName : list ${searchPortletList}) {
+				SearchPortlets._clickAddPortlet(portletName = ${portletName});
 
-		for (var portletName : list ${searchPortletList}) {
-			SearchPortlets._clickAddPortlet(portletName = ${portletName});
+				var key_portletName = ${portletName};
 
-			var key_portletName = ${portletName};
+				AssertElementPresent(locator1 = "Portlet#LOADED");
 
-			AssertElementPresent(locator1 = "Portlet#LOADED");
+				AssertElementPresent(locator1 = "Home#PORTLET");
+			}
+		}
+		else {
+			if (!(isSet(column))) {
+				var column = 1;
+			}
+
+			if (!(isSet(layoutName))) {
+				var layoutName = "Search";
+			}
 
-			AssertElementPresent(locator1 = "Home#PORTLET");
+			JSONLayout.addWidgetToPublicLayout(
+				column = ${column},
+				groupName = "Guest",
+				layoutName = ${layoutName},
+				widgetName = ${searchPortletList});
```

Comment:
Can we pass in a list of Search widgets for `JSONLayout.addWidgetToPublicLayout` instead of having to create `for` loop at the testcase level to pass add multiple widgets via JSON?